packages/react/src/r4b/provider.tsx

```diff
@@ -1,33 +1,26 @@
 import { Formatter } from "@bonfhir/core/r4b";
-import { PropsWithChildren, createElement, useMemo } from "react";
+import { PropsWithChildren, createElement } from "react";
 import { FhirUIContext, OnNavigateArgs } from "./context";
 import { FhirUIDefaultProps } from "./default-props";
+import { useFhirFormatters } from "./formatters/fhir-formatters-context";
+import { FhirFormattersProvider } from "./formatters/provider";
 import { FhirUIRenderer } from "./renderer";
 
-export type FhirUIProviderProps = PropsWithChildren<{
-  formatter?: Formatter | null | undefined;
+// DO NOT EXPORT, this is to support the independent formatters context
+type DeprecationWrapperProps = PropsWithChildren<{
   renderer: Partial<FhirUIRenderer>;
   defaultProps?: FhirUIDefaultProps | null | undefined;
   onNavigate?: (args: OnNavigateArgs) => void;
 }>;
 
-/**
- * Provide a context for ui components.
- * Should probably be placed near the top of your React app render tree.
- */
-export function FhirUIProvider(props: FhirUIProviderProps) {
-  const formatter = useMemo(() => {
-    if (props.formatter) {
-      return props.formatter;
-    }
-
-    return Formatter.default;
-  }, [props.formatter]);
+// DO NOT EXPORT, this is to support the independent formatters context
+const FhirUIProviderWrapper: React.FC<DeprecationWrapperProps> = (props) => {
+  const { formatter } = useFhirFormatters();
```

Comment:
Nitpick: Can we use a single noun everywhere around the formatter? I understand the utility composes a collection of type-specific formatters but the thing actually being used is a single instance of `Formatter`. I think we can also return just the formatter in that case and not an object with a `formatter` property. Unless I'm missing something, this could be:

```jsx
function Example() {
  const formatter = useFhirFormatter();

  return <>{formatter.format("HumanName", ...)}</>
}
```

Reply:
done! had to do some juggling with names since FhirFormatter is a react component